From 60f50c21151126dda3cf1bbc5e6ea76159a18a8c Mon Sep 17 00:00:00 2001
From: Timo Juhani Lindfors <timo.lindfors@iki.fi>
Date: Thu, 28 Feb 2013 17:15:43 +0200
Subject: [PATCH] blacklist more unsafe functions

These were found empirically. Some of the functions look innocent but
they are called from functions marked with __kprobes so they still
cause crashes. Further analysis is available at
https://lkml.org/lkml/2013/2/27/82 ("kprobing "hash_64.constprop.26"
crashes the system, recursion through get_kprobe?").
---
 dwflpp.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/dwflpp.cxx b/dwflpp.cxx
index 915e286..41a57c2 100644
--- a/dwflpp.cxx
+++ b/dwflpp.cxx
@@ -3424,6 +3424,11 @@ dwflpp::build_blacklist()
   blfn += "|test_ti_thread_flag.*";
   blfn += "|inat_get_opcode_attribute";
   blfn += "|system_call_after_swapgs";
+  blfn += "|HYPERVISOR_[gs]et_debugreg";
+  blfn += "|HYPERVISOR_event_channel_op";
+  blfn += "|hash_64";
+  blfn += "|hash_ptr";
+  blfn += "|native_set_pte";
 
   // Lots of locks
   blfn += "|.*raw_.*_lock.*";
-- 
1.7.2.5

